pyscriptjs/src/components/base.ts

```diff
@@ -304,16 +302,14 @@ export class PyWidget extends HTMLElement {
         this.wrapper = document.createElement('slot');
         this.shadow.appendChild(this.wrapper);
 
-        if (this.hasAttribute('src')) {
-            this.source = this.getAttribute('src');
-        }
-
-        if (this.hasAttribute('name')) {
-            this.name = this.getAttribute('name');
-        }
+        this.addAttributes('src','name','klass');
+    }
 
-        if (this.hasAttribute('klass')) {
-            this.klass = this.getAttribute('klass');
+    addAttributes(...attrs:string[]){
+        for (const each of attrs){
```

Comment:
This is overwriting `this.source` every time this has the `each` attribute (which is true for `name` and `klass`). It should assign `each` as an attribute of `this` and handle renaming differently. Maybe through a mapping (optional) parameter?

Reply:
Oops, you're right. Should be fixed with latest commit